specification/vmware/resource-manager/Microsoft.AVS/preview/2021-01-01-preview/vmware.json

```diff
@@ -0,0 +1,5545 @@
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/DatastoreList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Datastores_List": {
+            "$ref": "./examples/Datastores_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/clusters/{clusterName}/datastores/{datastoreName}": {
+      "get": {
+        "operationId": "Datastores_Get",
+        "tags": [
+          "Datastores"
+        ],
+        "summary": "Get a datastore in a private cloud cluster",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/privateCloudName"
+          },
+          {
+            "$ref": "#/parameters/clusterName"
+          },
+          {
+            "$ref": "#/parameters/datastoreName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/Datastore"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Datastores_Get": {
+            "$ref": "./examples/Datastores_Get.json"
+          }
+        }
+      },
+      "put": {
+        "operationId": "Datastores_Create",
+        "x-ms-long-running-operation": true,
+        "tags": [
+          "Datastores"
+        ],
+        "summary": "Create a datastore in a private cloud cluster",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/privateCloudName"
+          },
+          {
+            "$ref": "#/parameters/clusterName"
+          },
+          {
+            "$ref": "#/parameters/datastoreName"
```

Comment:
Is this required even when we have the section below? ... Also does the same need to be done for GET/DELETE datastore resource as well?

Reply:
Yes, `datastoreName` is required for `Datastores_Creates`, `Datastores_Get`, & `Datastores_Delete`.  `Datastores_Create` is the only one that requires a body.